dataprovider/k8s_data_provider.go

```diff
@@ -0,0 +1,118 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dataprovider
+
+import (
+	"context"
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/resources/providers"
+	"github.com/elastic/cloudbeat/version"
+	"github.com/elastic/elastic-agent-autodiscover/kubernetes"
+	"github.com/elastic/elastic-agent-libs/logp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+type k8sDataCollector struct {
+	kubeClient k8s.Interface
+	log        *logp.Logger
+	cfg        *config.Config
+}
+
+type k8sDataProvider interface {
+	CollectK8sData(ctx context.Context) CommonK8sData
+}
+
+func NewK8sDataProvider(log *logp.Logger, cfg *config.Config) k8sDataProvider {
+	kubeClient, err := providers.KubernetesProvider{}.GetClient(log, cfg.KubeConfig, kubernetes.KubeClientOptions{})
+	if err != nil {
+		log.Errorf("NewK8sDataProvider error in GetClient: %v", err)
+	}
+
+	return k8sDataCollector{
+		kubeClient: kubeClient,
+		log:        log,
+		cfg:        cfg,
+	}
+}
+
+func (k k8sDataCollector) CollectK8sData(ctx context.Context) CommonK8sData {
```

Comment:
Let's return a pointer, and return a `nil` instead of empty struct when k8s is not available